File: Railway Management System Project/spring_backend_project/src/main/java/com/app/service/TrainService.java
package com.app.service;

import java.util.List;

import com.app.dto.TrainDTO;
import com.app.dto.TrainOnlyDTO;
import com.app.dto.TrainRescheduleDTO;
import com.app.dto.TrainSrcDestDateDTO;
import com.app.entities.Train;

public interface TrainService {
	
	Train addTrain(TrainDTO trainDTO);

	TrainOnlyDTO findTrainByNumber(Long trainNumber);

	TrainOnlyDTO findTrainByName(String trainName);

	List<TrainOnlyDTO> findAllTrains();

	List<TrainDTO> findTrainBySourceDestinationDate(TrainSrcDestDateDTO srcDestDate);

	TrainDTO findTrainById(Long trainId);

	List<TrainOnlyDTO> getAllTrainsByAdmin(Long adminId);

	List<TrainDTO> findTrainsByTwoStopsInSequence(TrainSrcDestDateDTO searchInfo);

	String reschuduleTrain(TrainRescheduleDTO trainRescheduleDTO);
}

File: Railway Management System Project/client/src/Services/train.service.js
import httpClient from '../http-common';

const rescheduleTrain = (data) => {
    return httpClient.put('/trains//reschedule', data);
};

const findTrainBySourceDestinationDate = (data) => {
    return httpClient.post('/trains/searchTrain', data);
};

const findTrainsByTwoStopsInSequence = (data) => {
    return httpClient.post('/trains/searchTrain/startAndStop', data);
};

const addTrain = (data) => {
    return httpClient.post('/trains/add', data);
};

const findTrainAllDetails = (trainId) => {
    return httpClient.get(`/trains/searchTrainAllDetails/${trainId}`);
};

const findTrainByNumber = (trainNumber) => {
    return httpClient.get(`/trains/number/${trainNumber}`);
};

const findTrainByName = (trainName) => {
    return httpClient.get(`/trains/name/${trainName}`);
};

const findTrainDetailsById = (trainId) => {
    return httpClient.get(`/trains/findTrainById/${trainId}`);
};

const getAllTrains = () => {
    return httpClient.get('/trains/allTrains');
};

const getAllTrainsByAdmin = (adminId) => {
    return httpClient.get(`/trains/admin/allTrain/${adminId}`);
};

export default { rescheduleTrain, findTrainBySourceDestinationDate, findTrainsByTwoStopsInSequence, addTrain, findTrainAllDetails, findTrainByNumber, findTrainByName, findTrainDetailsById, getAllTrains, getAllTrainsByAdmin };
